### PR TITLE
Add sk console gpt

### PR DESCRIPTION
**Pull request by jimbobbennett**
### Motivation and Context
Adding a console version of ChatGPT at the request of @markwallace-microsoft.

### Description
This adds a new C# starter sample that provides a basic implementation of a chat GPT clone, leverage chat completion, and chat history for memory.

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel-starters/blob/main/CONTRIBUTING.md)
- [X] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: sk-csharp-console-chat/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+dotnet user-secrets set "apiKey" "... your OpenAI key ..."
+```
+
+Configure an Azure OpenAI endpoint
+
+```bash
+cd sk-csharp-console-chat
+dotnet user-secrets set "serviceType" "AzureOpenAI"
+dotnet user-secrets set "serviceId" "gpt-3.5-turbo"
+dotnet user-secrets set "deploymentOrModelId" "gpt-3.5-turbo"
```
Comment (markwallace-microsoft) [inline sk-csharp-console-chat/README.md:39]: There is a naming inconsistency between Azure OpenAI and OpenAI.

Azure uses `gpt-35-turbo` and not `gpt-3.5-turbo`

File: sk-csharp-console-chat/config/appsettings.json.azure-example [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "serviceType": "AzureOpenAI",
+  "serviceId": "gpt-3.5-turbo",
+  "deploymentOrModelId": "gpt-3.5-turbo",
```
Comment (markwallace-microsoft) [inline sk-csharp-console-chat/config/appsettings.json.azure-example:4]: Should be `gpt-35-turbo`

**Review by markwallace-microsoft — COMMENTED** [review]
Couple of small changes

**Comment by jimbobbennett** [comment]
> Couple of small changes

Fixed!

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**